controllers/datadogagent/feature/admissioncontroller/feature.go

```diff
@@ -308,6 +354,28 @@ func (f *admissionControllerFeature) ManageClusterAgent(managers feature.PodTemp
 			})
 		}
 
+		if f.agentSidecarConfig.selectors != nil {
```

Comment:
I think here you want to check for length, not `nil`
```suggestion
		if len(f.agentSidecarConfig.selectors) > 0 {
```